healthz/README.md

```diff
@@ -7,41 +14,239 @@ in distributed systems.
 The ability to also include "implementation specific" details about the health
 of the system or component is very standard for network devices in the form for
 "show tech", "diag" or "debug" commands. These are very useful for getting
-useful diagnostics about the system state to the implementers. Healthz exposes
+diagnostics about the system state to the implementers. Healthz exposes
 these interfaces as queriable endpoints to allow operators to validate "health"
-on components and if "unhealthy" gather implementor specific data to help triage
-or reproduce issues.
-
-## Design
-
-* The service definition provides a Get method which is correlated with a
-  /component in the OC tree. The Get method is responsible for getting the
-  current status of the requested component path. This will cause the system
-  to fetch the current state of component this may entail running specific
-  validations and gathering any relevant status of the component to reflect
-  the "health check"
-
-* The internal design details of the "health check" and the frequency of the
-  health check are up to the implementor to determine. This presentation of
-  the status in OC is proposed to be modeled as
-
-```yang
-component/healthz/status = Enum
-component/healthz/last-unhealthy = Timestamp
-component/healthz/unhealthy_count = Int64
+on components and if "unhealthy" gather implementor specific data to help
+triage or reproduce issues.
+
+## Design Background
+
+`Healthz` provides a means by which a user may initiate checks of health
+(through the `Check` RPC) or a system may report the results of a check that it
+has initiated of its own accord. 
```

Comment:
re "a system may report the results of a check that it has initiated of its own accord.":

1. when a system boots, should the relevant component manager perform initial health checks and thus generate ComponentStatus for components? 
2. When a certain component changes its health status from one state to another (think a tranceiver goes from healthy to unhealthy), should the relevant component manager record a ComponentStatus event with a distinct ID automatically?

Reply:
i would say it should look like this:

system boots
gnmi tree is initialized for things which have not run a check yet - they start in the unspecfied state.
During initialization of the subcomponents the system should perform initial health checks to put the components into a known state 
if the component is unhealthy - generate an event
if the components is healthy - set state and move on

2. yes if something goes unhealthly there should be an event - but how much work is done to "triage" that event is left to the vendor.